src/Viewer/PageViewer.jsx

```diff
@@ -365,41 +377,128 @@ export default function PageViewer(props) {
     baseVariableValue,
     componentName,
     rendererType,
+    promiseResolve,
   }) {
+    // Note: the reason we check both the renderer class and .type
+    // is that if the renderer was memoized, then the renderer class itself in on .type,
+    // Otherwise, the renderer class is the value of the rendererClasses entry.
     let ignoreActionsWithoutCore =
       rendererClasses.current[rendererType]?.ignoreActionsWithoutCore ||
       rendererClasses.current[rendererType]?.type?.ignoreActionsWithoutCore;
-    if (coreCreated.current || !ignoreActionsWithoutCore?.(action.actionName)) {
-      let actionId = nanoid();
-      args = { ...args };
-      args.actionId = actionId;
+    if (
+      !coreCreated.current &&
+      (ignoreActionsWithoutCore?.(action.actionName) ||
+        !coreCreationInProgress.current)
+    ) {
+      // The action is being skipped because core has not been created
+      // and either the action must be ignored without core or core isn't actually
+      // in the process of being created (relevant case is that core has been terminated)
+
+      if (promiseResolve) {
+        // if were given promiseResolve, then action was called from resolveAction
+        // where the return value is being ignored.
+        // Resolve promise as false as action will be skipped
+        promiseResolve(false);
+        return;
+      } else {
+        // Action was called normally where it is expecting a promise to be returned.
+        // A promise resolved as false indicates action was skipped.
+        return Promise.resolve(false);
+      }
+    }
 
-      if (baseVariableValue !== undefined && componentName) {
-        updateRendererUpdatesToIgnore({
-          coreId: coreId.current,
-          componentName,
+    if (lastSkippedAction.current) {
+      // we are for sure skipping the lastSkippedAction,
```

Comment:
Still working to grok everything going on here, but this language makes it seem like the variable name might be a little misleading. I assume that resolving the promise with false is what is needed to  "really" say it was skipped. So what are the conditions where this "lastSkippedAction" won't be skipped? Might be more awkward, but should this variable be called something like lastSkippableAction?

Reply:
Right, it is misleading.  I'm renaming it to actionTentativelySkipped, as "last" doesn't make sense in that context given that only one is *tentatively* skipped.